Add odk-new-repeat event

lognaturel opened the pull request:
Closes #246 

Based on the conversation at https://forum.opendatakit.org/t/xforms-spec-proposal-add-event-fired-when-new-repeat-instance-is-created/19616

Please let me know if it's feeling like too much detail. I tried to capture everything we discussed and leave as little ambiguity as possible. In particular, the contrasting with W3C XForms feels like useful context to me but it's not really necessary.

The example is not realistic -- why in the world would 2 above the user's age be a useful default? My goal is to show an expression and the evaluation context for `value`. If a reviewer thinks a more realistic example would help, I can think of something.

CC @tiritea

lognaturel committed:
Add odk-new-repeat event

MartijnR commented on _sections/65-events-and-actions.md:
```diff
@@ -12,9 +12,40 @@ See the W3C XForms specification [section on events](https://www.w3.org/TR/xform
 
+#### The `odk-new-repeat` event
+The `odk-new-repeat` event is dispatched when a new instance of a repeat is added to the primary instance and before recomputation of `calculates`, `constraints`, etc. Actions triggered by `odk-new-repeat` must be nested in the repeat form control.
+
```
The paragraph below is indeed very good and useful to record for posterity, but I don't think it should be in the spec. Maybe we can at least paste it into the GitHub issue?

lognaturel replied:
👍

MartijnR commented on _sections/65-events-and-actions.md:
```diff
@@ -12,9 +12,40 @@ See the W3C XForms specification [section on events](https://www.w3.org/TR/xform
 
 *Note: `xforms-ready` was previously documented as the event dispatched the first time an instance is loaded. Since that definition does not match the W3C XForms event with the same name, it was deprecated in favor of `odk-instance-first-load`.*
 
+#### The `odk-new-repeat` event
```
maybe leave out the quotation marks for odk-new repeat as it looks a little odd in the HTML page?

MartijnR commented on _sections/65-events-and-actions.md:
```diff
@@ -12,9 +12,40 @@ See the W3C XForms specification [section on events](https://www.w3.org/TR/xform
 | --------------------------| ----------- |
 | <a id="event:odk-instance-first-load" href="#event:odk-instance-first-load">`odk-instance-first-load`</a><a id="event:xforms-ready"></a>            | dispatched the first time an instance is loaded |
 | <a id="event:xforms-value-changed" href="#event:xforms-value-changed">`xforms-value-changed`</a>    | dispatched after an instance data node's value changes. |
+| <a id="event:odk-new-repeat" href="#event:odk-new-repeat">`odk-new-repeat`</a>	| dispatched when a new instance of a repeat is added to the primary instance and before recomputation of `calculates`, etc. Actions triggered by `odk-new-repeat` must be nested in the repeat form control. <a href="#the-odk-new-repeat-event">See more</a>.
```
I think this can be kept short without the 
> and before recomputation of `calculates`, etc. Actions triggered by `odk-new-repeat` must be nested in the repeat form control

Since that is mentioned right afterwards

lognaturel replied:
Ah yes, good point. 😊

MartijnR commented:
Thanks! This is very good. Sorry, I somehow missed the alert email for this PR.

lognaturel committed:
Review feedback [please squash]

lognaturel commented:
Thank you! I addressed your feedback in a separate commit so the diff was clear but please squash. And do let me know if you see anything else worth improving.
